pex/pex_info.py

```diff
@@ -527,12 +540,14 @@ def update(self, other):
                 other.interpreter_constraints
             )
         self._requirements.update(other.requirements)
+        self._excluded.update(other.excluded)
```

Comment:
Is the behaviour when `self` includes something excluded by `other` (or vice versa) something that's been explicitly considered?

It looks like this code will consider those requirements as "excluded". Is that intentional or accidental? If the latter, is it desirable?

Reply:
It has, but not in this context. There is actually a test for the case when the same PexInfo includes contradictory root requirements and excludes. That case warns at build time and is silent at run time. This fits my general philosophy of trying not to piss off an advanced user (the warning is even questionable in my mind). As the semantics stand, a user with a batch system that creates these command lines can have excludes trump requirements and they just get a warning. If they want this to trigger a fail-fast error, they can add that behavior in their code that generates the command line. If I went with the nanny bias, however, they would be no such recovery for the aforementioned batch user. They would be forced to curse and submit a patch adding some new option to relax the check.

Reply:
After more thought, I'm even less convinced the warning is a good idea. The obvious use case is using Pex to package but some other means to lock. In other words, something like `pex -r requirements.lock --exclude build --exclude my --exclude subset ...`. Some other tool provides the lock in Pip requirements style (hopefully with hashes), but you can still use Pex to lock a subset - here by passing `locked-reqs - desired-reqs` as the `--exclude`s. The reachability behavior documented for `--exclude` is what builds the proper subset.

Since the warning can be reverted without breaking people though, I'll proceed as-is and see if the the use case I realized even comes up / comes up enough to get complaints about the warnings / having to `--no-emit-warnings`.